src/averageAges.js

```diff
@@ -1,61 +1,52 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const filtered = people.filter(century > 0
+    ? person => century === Math.ceil(person.died / 100) && person.sex === 'm'
+    : person => person.sex === 'm');
+
+  const averageAge = filtered.reduce((prev, person) => {
+    const age = person.died - person.born;
+
+    return prev + age;
+  }, 0)
+  / filtered.length;
+
+  return averageAge;
 }
 
-/**
- * Implement calculateWomenAverageAge function
- *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
- *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
- *
- * @param {object[]} people
- * @param {boolean} withChildren - optional
- *
- * @return {number}
- */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filtered = people.filter(withChildren !== undefined
```

Comment:
what if someone will pass a false here 
<img width="210" alt="image" src="https://github.com/mate-academy/js_average-ages/assets/70220667/d50a2720-fb55-4e79-b9e8-1924b740936f">
```suggestion
  const filtered = people.filter(!!withChildren
```